nixos/modules/tasks/network-interfaces-systemd.nix

```diff
@@ -12,7 +12,13 @@ let
     i.ipv4.addresses
     ++ optionals cfg.enableIPv6 i.ipv6.addresses;
 
-  dhcpStr = useDHCP: if useDHCP == true || useDHCP == null then "both" else "none";
+  dhcpStr = useDHCP:
+    if useDHCP == true || useDHCP == null
+    then
+      if cfg.enableIPv6
+      then "both"
```

Comment:
I am (and never was, so far) happy with how this worked.

Let me quote the `DHCP=` option from `systemd.network`:

        DHCP=
           Enables DHCPv4 and/or DHCPv6 client support. Accepts "yes", "no",
           "ipv4", or "ipv6".

           Note that DHCPv6 will by default be triggered by Router Advertisement,
           if that is enabled, regardless of this parameter. By enabling DHCPv6
           support explicitly, the DHCPv6 client will be started regardless of the
           presence of routers on the link, or what flags the routers pass. See
           "IPv6AcceptRA=".

The painpoint for me is the fact that we will enable DHCPv6. DHCPv6 is not the default way of configuring IPv6. It will not hurt (most people, anyway).

I would prefer having an option to specify the IPv6 DHCP configuration on it's own. 

What do you think of having `useDHCP` which enables two (new) options: `useDHCPv4` & `useDHCPv6`? Whoever cares about such details (like me) can then use the specific protocol switch to configure it.

Or, maybe just allow `useDHCP` to be an `types.nullOr enum [ "off" "ipv6" "ipv4" "both" ];`?

Opinions?

Reply:
Very good observation indeed. I think I got confused because `both` is not a valid option....

 My goad writing this was to only not enable DHCPv6 if it was not strictly needed. What I needed in my case, was to use `DHCP=ipv4` so that there was that the interface would get stuck in  the configuring state.

I don't think the use of `networking.enableIPv6 = false` is mainstream, the only reason I used it is to work around problems like https://github.com/coreos/bugs/issues/1419.